lib/Dialect/FIRRTL/FIRRTLTypes.cpp

```diff
@@ -586,6 +586,28 @@ bool firrtl::areTypesWeaklyEquivalent(FIRRTLType destType, FIRRTLType srcType,
          destFlip == srcFlip;
 }
 
+/// Returns true if the destination is at least as wide as an equivalent source.
+bool firrtl::isTypeLarger(FIRRTLType dstType, FIRRTLType srcType) {
+  return TypeSwitch<FIRRTLType, bool>(dstType)
+      .Case<BundleType>([&](auto dstBundle) {
+        auto srcBundle = srcType.cast<BundleType>();
+        for (size_t i = 0, n = dstBundle.getNumElements(); i < n; ++i)
+          if (!isTypeLarger(dstBundle.getElement(i).type,
+                            srcBundle.getElement(i).type))
```

Comment:
should this be flip aware?
e.g. for Bundle types like this, maybe we should switch `dst` and `src`?
```
wire a : { flip a: UInt<1> }
wire b : { flip b: UInt<8> }
a <= b
```

Reply:
Good catch!